Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -398,12 +399,32 @@ bool IsLinux()
 #endif
             }
 
-            bool IsWindowsGoogleBios() => throw new NotImplementedException();
+            bool IsWindowsGoogleBios()
+            {
+                Logger.Info("Checking BIOS values on Windows.");
+#if NET45 || NETSTANDARD1_3
+                Logger.Debug("System.Management is not supported in net45 and netstandard1.3");
+                return false;
+#else
+                System.Management.ManagementClass biosClass = new ("Win32_BIOS");
+                using var instances = biosClass.GetInstances();
+
+                foreach(var instance in instances) using (instance)
+                {
+                    // We should only find one instance for Win32_BIOS class.
+                    // This will throw a ManagementException "NotFound" if the "Manufacturer" property
```

Comment:
But if there are multiple instances, it could be present in one other than the first one, no?

Reply:
Yes, in principle I guess it could happen. We are now iterating over all instances.